test/tools/syntaxTestTool.cpp

```diff
@@ -0,0 +1,354 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <test/libsolidity/AnalysisFramework.h>
+#include <test/libsolidity/SyntaxTestParser.h>
+#include <boost/algorithm/string/replace.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <iostream>
+#ifdef __unix__
+#include <unistd.h>
+#include <termios.h>
+#endif
+
+using namespace dev::solidity;
+using namespace dev::solidity::test;
+using namespace std;
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+namespace color
```

Comment:
WIP: put this some better place than a global namespace

Reply:
In general color coding the output improves the usability of the tool a lot, though.

Reply:
Yay, colours! :)